src/index/generateTrees/shared/findSharedParent.ts

```diff
@@ -4,17 +4,18 @@ import path from "path";
 export const findSharedParent = (paths: string[]) => {
   if (paths.length === 1) return path.dirname(paths[0]);
 
-  const parentPaths: string[] = [];
-  const parts: string[][] = paths.map(x => x.split("/"));
-  const firstParts = parts[0];
+  const [shortest, secondShortest] =
+    paths.length > 2 ? paths.sort((a, b) => a.length - b.length) : paths;
 
-  firstParts.forEach((part, idx) => {
-    const allPartsMatch = parts.every(p => p[idx] === part);
+  const secondShortestParts = secondShortest.split("/");
+  const parentPaths: string[] = [];
 
-    if (allPartsMatch) {
+  let index = 0;
+  for (const part of shortest.split("/")) {
+    if (part === secondShortestParts[index++]) {
       parentPaths.push(part);
     }
-  });
+  }
```

Comment:
Why not using a `forEach`? Is it faster?

Reply:
Performance wise `for` loop is more interesting. https://jsperf.com/loops

Reply:
@SimonJang I'd currently prefer readability before anything else currently. We do not have any noticeable performance issues.

Furthermore, there is a big difference in performance in a simple index based `for` loop and a `forof` loop. A `forof` loop is actually slower than a `forEach` loop.